hub/apps/get-started/uno-simple-photo-viewer.md

```diff
@@ -0,0 +1,931 @@
+
+```csharp
+private void ImageGridView_ContainerContentChanging(
+ListViewBase sender,
+ContainerContentChangingEventArgs args)
+{
+
+    if (args.InRecycleQueue)
+    {
+        var templateRoot = args.ItemContainer.ContentTemplateRoot as Grid;
+        var image = templateRoot?.FindName("ItemImage") as Image;
+        if (image is not null)
+        {
+            image.Source = null;
+        }
+    }
+
+#if WINDOWS
+        if (args.Phase == 0)
+        {
+            args.RegisterUpdateCallback(ShowImage);
+            args.Handled = true;
+        }
+#else
+    ShowImage(sender, args);
+#endif
+}
+```
+
+The specialized callback is now only registered using `ContainerContentChangingEventArgs.RegisterUpdateCallback()` if the platform is Windows. Otherwise, the `ShowImage` method is called directly. We will also need to make changes to the `ShowImage` method to work alongside the changes made to the `ImageGridView_ContainerContentChanging` method. Replace the `ShowImage` method in the `MainPage.xaml.cs` file with the following code:
+
+```csharp
+private async void ShowImage(ListViewBase sender, ContainerContentChangingEventArgs args)
+{
+    if (
+#if WINDOWS
+            args.Phase == 1
+#else
+        true
+#endif
+        )
+    {
+
+        // It's phase 1, so show this item's image.
+        var templateRoot = args.ItemContainer.ContentTemplateRoot as Grid;
+        var image = templateRoot?.FindName("ItemImage") as Image;
+        var item = args.Item as ImageFileInfo;
+#if WINDOWS
+        if (image is not null && item is not null)
+        {
+            image.Source = await item.GetImageThumbnailAsync();
+        }
+#else
+        if (item is not null)
+        {
+            await item.GetImageSourceAsync();
+        }
+#endif
+    }
+}
+```
+
+Again, preprocessor directives ensure that the `ContainerContentChangingEventArgs.Phase` property is only used where it is supported. We make use of the previously unused `GetImageSourceAsync()` method to load the image files into the `GridView` on platforms other than Windows. At this point, we will accommodate the changes made above by editing to the `ImageFileInfo` class. 
+
+### Creating a separate code path for other platforms
+
+Update `ImageFileInfo.cs` to include a new property called `ImageSource` that will be used to load the image file.
+
+```csharp
+public BitmapImage? ImageSource { get; private set; }
+```
+
+Because platforms like the Web do not support advanced image file properties that are readily available on Windows, we should add a constructor overload which does not require an `ImageProperties` typed parameter. Add a new overload under the existing one using the following code:
```

Comment:
```suggestion
Because platforms like the Web do not support advanced image file properties that are readily available on Windows, we should add a constructor overload that does not require an `ImageProperties` typed parameter. Add a new overload under the existing one using the following code:
```